datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -0,0 +1,116 @@
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.Flow2Handler;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.service.flowlog.FlowChainLogService;
+import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
+import com.sequenceiq.sdx.api.model.SdxRecoveryRequest;
+import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
+
+@Component
+/**
+ * Provides entrypoint for recovery of failed SDX resize.
+ *
+ * The main entry point is {@code triggerRecovery}, which starts a cloudbreak Flow to recover the Data Lake.
+ * To ensure a Resize recovery is appropriate, use one of {@code canRecover} or {@code validateRecovery}.
+ */
+public class ResizeRecoveryService implements RecoveryService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResizeRecoveryService.class);
+
+    @Inject
+    private SdxService sdxService;
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private SdxReactorFlowManager sdxReactorFlowManager;
+
+    @Inject
+    private Flow2Handler flow2Handler;
+
+    @Inject
+    private FlowChainLogService flowChainLogService;
+
+    /**
+     * Determines if it's possible to run a <em>Resize</em> recovery, returning a simple boolean indicator
+     *
+     * @param crn       the CRN of the user requesting recovery
+     * @param nameOrCrn the name or CRN of the resource to recover
+     * @return true if resize recovery is possible.
+     */
+    public boolean canRecover(String crn, NameOrCrn nameOrCrn) {
+        SdxCluster cluster = sdxService.getByNameOrCrn(crn, nameOrCrn);
+        FlowLog flowLog = flow2Handler.getFirstStateLogfromLatestFlow(cluster.getId());
+        if (DatalakeResizeFlowEventChainFactory.class.getSimpleName().equals(flowChainLogService.getFlowChainType(flowLog.getFlowChainId()))) {
+            SdxStatusEntity actualStatusForSdx = getSdxStatusEntity(crn, nameOrCrn);
+            if (actualStatusForSdx.getStatus().equals(DatalakeStatusEnum.STOP_FAILED)) {
+
+                return true;
+            }
+
+        }
+        return false;
+    }
+
+    private SdxStatusEntity getSdxStatusEntity(String crn, NameOrCrn nameOrCrn) {
+        SdxCluster cluster = sdxService.getByNameOrCrn(crn, nameOrCrn);
+        return sdxStatusService.getActualStatusForSdx(cluster);
+    }
+
+    /**
+     * Determines if it's possible to run a <em>Resize</em> recovery, returning a detailed validation message.
+     * <p>
+     * This is distinguished from {@code canRecover} by including a detailed validation message, usable as a web response.
+     *
+     * @param crn       the CRN of the user requesting recovery
+     * @param nameOrCrn the name or CRN of the resource to recover
+     * @return detailed validation message, as a web response
+     */
+    public SdxRecoverableResponse validateRecovery(String crn, NameOrCrn nameOrCrn) {
```

Comment:
This depends on the answer to my comments above, but if possible it would be nice to have this just call the **canRecover** method in order to simplify and centralize this logic.

Reply:
I think one possible way to express the difference I want to highlight might be to add a RecoveryStatus.NOT_APPLICABLE to make it clear why something can't be recovered

Reply:
I'm not sure of the other changes in scope required to expand that enum

Reply:
Yeah I think it would be helpful to make it very clear what these two methods are for. When I first read through your changes it was very hard to tell why they differed.